From 1a6017fd9185b88f252279ddab34a324d8f5cb57 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Jan-Simon=20M=C3=B6ller?= <dl9pf@gmx.de>
Date: Tue, 20 Nov 2012 21:22:18 +0100
Subject: [PATCH 1005/1005] slab-builtin_contant_p.patch

Hack: silence warning/error on builtin_constant_p.

Use of __builtin_constant_p(size) breaks Clang compile

Do not upstream this patch as is.

Signed-off-by: Mark Charlebois <charlebm@gmail.com>

---
 mm/slab.c |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/mm/slab.c b/mm/slab.c
index 33d3363..0cfe5c8 100644
--- a/mm/slab.c
+++ b/mm/slab.c
@@ -342,7 +342,8 @@ static __always_inline int index_of(const size_t size)
 {
 	extern void __bad_size(void);
 
-	if (__builtin_constant_p(size)) {
+	/* if (__builtin_constant_p(size)) { */
+	if (1) {
 		int i = 0;
 
 #define CACHE(x) \
-- 
1.7.10.4

